README.md

```diff
@@ -78,6 +78,117 @@ createAction('ADD_TODO')('Use Redux');
 
 `metaCreator` is an optional function that creates metadata for the payload. It receives the same arguments as the payload creator, but its result becomes the meta field of the resulting action. If `metaCreator` is undefined or not a function, the meta field is omitted.
 
+### `createActions(?actionMap, ?arrayDefTypes, ?optionMap)`
+
+Create multiple actions and return an object mapping from the type to the actionCreator. This eliminates some of the repetitiveness when creating many actions.
+
+If `actionMap` is provided as first argument, an action will be created for each item in the map. The key will be used for the type. The value will be treated as follows:
+
+ - if value is a function then it will be assumed to be the payloadCreator function
+ - if value is an object then it will accept as its keys, `payload` and `meta`, both of which are functions that will serve as the payloadCreator and/or metaCreator. If payloadCreator is missing the default payloadCreator is used. If metaCreator is missing then it will not use one.
+ - otherwise create a default actionCreator for the key type
+
+If an array of string is provided as `arrayDefTypes` it will be used to create default actionCreators using the string for the action type.
+
+If an `optionMap` object is supplied as last argument then it will adjust the options createActions uses in operation:
+
+ - `prefix` - string, prefix all action types with this prefix, default ''
+
+#### Example: simple default actions
+
+```js
+import { createActions } from 'redux-actions';
+const a = createActions(['ONE', 'TWO', 'THREE']);
+
+// is equivalent to
+const a = {
+  ONE: createAction('ONE'),
+  TWO: createAction('TWO'),
+  THREE: createAction('THREE')
+;}
+```
+
+#### Example: simple default actions with prefix
+
+```js
+import { createActions } from 'redux-actions';
+const a = createActions(['ONE', 'TWO', 'THREE'], { prefix: 'ns/');
+
+// is equivalent to
+const a = {
+  ONE: createAction('ns/ONE'),
+  TWO: createAction('ns/TWO'),
+  THREE: createAction('ns/THREE')
+;}
+```
+
+#### Example: using actionMap
+
+```js
+import { createActions } from 'redux-actions';
+const a = createActions({
+  ONE: (b, c) => ({ bar: b, cat: c }),  // payload creator
+  TWO: {
+    payload: (d, e) => [d, e]            // payload creator
+    meta: (d, e) => ({ time: Date.now() }) // meta creator
```

Comment:
I think it is important to convey that the meta creator actually get the
same variables that one can use if they need to. The docs were not clear on
this so I had to look at the code to see whatparams they receive.
Commenting them out indicates they are there but not used and this will
pass eslint validation.

On Sunday, July 17, 2016, Victor Alvarez <notifications@github.com> wrote:

> In README.md
> <https://github.com/acdlite/redux-actions/pull/104#discussion_r71076148>:
>
> > +const a = {
> > +  ONE: createAction('ns/ONE'),
> > +  TWO: createAction('ns/TWO'),
> > +  THREE: createAction('ns/THREE')
> > +;}
> > +```
> > +
> > +#### Example: using actionMap
> > +
> > +```js
> > +import { createActions } from 'redux-actions';
> > +const a = createActions({
> > +  ONE: (b, c) => ({ bar: b, cat: c }),  // payload creator
> > +  TWO: {
> > +    payload: (d, e) => [d, e]            // payload creator
> > +    meta: (d, e) => ({ time: Date.now() }) // meta creator
>
> Do you need (d, e)? Does it help to convey the meaning of this example
> snippet?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/acdlite/redux-actions/pull/104/files/628cf27017ee8dac86e5f1b08606b826e753e056#r71076148>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAWOQDZNSyvyhuXjphiKnZev0-RLyk8ks5qWap7gaJpZM4JIZ-x>
> .
>


-- 
Jeff Barczewski
Founder of CodeWinds
http://codewinds.com/
Online developer training